dot/state/epoch.go

```diff
@@ -399,7 +400,8 @@ func (nem nextEpochMap[T]) Retrieve(blockState *BlockState, epoch uint64, header
 		// sometimes while moving to the next epoch is possible the header
 		// is not fully imported by the blocktree, in this case we will use
 		// its parent header which migth be already imported.
-		if errors.Is(err, blocktree.ErrEndNodeNotFound) {
+		if errors.Is(err, blocktree.ErrEndNodeNotFound) || errors.Is(err, trieproof.ErrKeyNotFound) ||
+			errors.Is(err, blocktree.ErrKeyNotFound) || errors.Is(err, chaindb.ErrKeyNotFound) {
```

Comment:
Can the returned error be one of these 4 sentinel errors?? I don't think the `trieproof` code is called anywhere in `blockState.IsDescendantOf(hash, headerHash)` for example.

Reply:
resolved